utils/scan_parameters.py

```diff
@@ -106,70 +118,84 @@ def ExtractFromXML(df, key, report_file):
             path = path.find(i)
         df.at[key, k] = path.text
 
-def ExtractROC(df, key, output_filename):
-    truth_filename = "../data_to_txt_2NN/KERAS_check_best_model_truth_labels.dat"
-    predict_filename = "../data_to_txt_2NN/KERAS_check_best_model_predictions.dat"
+def ExtractROC(df, key, output_filename,keras_dir):
+    truth_filename = "%s/KERAS_check_best_model_truth_labels.dat"%(keras_dir)
+    predict_filename = "%s/KERAS_check_best_model_predictions.dat"%(keras_dir)
 
     if "AUC" not in df:
         df["AUC"] = None
 
+    if "ExpAUC" not in df:
+        df["ExpAUC"] = None
+
     ## Check file exits
     if not os.path.isfile(output_filename):
         df.at[key, "AUC"] = None
+        df.at[key, "ExpAUC"] = None
+        return False
+
+    ## Check csim passed w/o errors
+    if not PassCSim(key):
+        df.at[key, "AUC"] = None
+        df.at[key, "ExpAUC"] = None
         return False
 
-    truth_df = pandas.read_csv(truth_filename, header=None)
-    predict_df = pandas.read_csv(predict_filename, header=None)
-    output_df = pandas.read_csv(output_filename, header=None)
+    truth_df = np.loadtxt(truth_filename)
+    predict_df = np.loadtxt(predict_filename)
+    output_df = np.loadtxt(output_filename)
```

Comment:
In case of multiple output, the returned numpy array is a matrix with shape (Nevents,Nclasses), where Nevents = number of rows, and Nclasses = number of columns of the input txt file. If your file has only one column because the NN output is just one, then Nclasses=1 and everything else should keep working.